### PR TITLE
Add NavigationBar dartpad example 

### PR DESCRIPTION
**Pull request by maheshj01**
This PR adds a dartpad example for Material 3 [NavigationBar](https://api.flutter.dev/flutter/material/NavigationBar-class.html)

Addresses https://github.com/flutter/flutter/issues/96750

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 1]
```diff
@@ -33,32 +33,19 @@ import 'tooltip.dart';
...
-///         icon: Icon(Icons.bookmark_border),
-///         label: 'Saved',
-///       ),
-///     ],
-///   ),
-/// ),
-/// ```
+/// {@tool dartpad}
+/// This example shows a [NavigationBar] as it is used within a [Scaffold]
+/// widget. The [NavigationBar] has three [NavigationDestination] widgets,
```
Comment (2002Bishwajeet) [inline packages/flutter/lib/src/material/navigation_bar.dart:38]: the comma should be ommitted

Comment (maheshmnj) [reply packages/flutter/lib/src/material/navigation_bar.dart]: Looks like comma isn't the issue

Comment (2002Bishwajeet) [reply packages/flutter/lib/src/material/navigation_bar.dart]: umm, I was actually talking about the grammar here😅. Which you have fixed now

Comment (maheshmnj) [reply packages/flutter/lib/src/material/navigation_bar.dart]: The checks failure looks related to https://github.com/flutter/flutter/issues/89243

**Comment by maheshmnj** [comment]
The checks failure looks related to https://github.com/flutter/flutter/issues/89243

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 2]
```diff
@@ -33,32 +33,19 @@ import 'tooltip.dart';
...
-///   ),
-/// ),
-/// ```
+/// {@tool dartpad}
+/// This example shows a [NavigationBar] as it is used within a [Scaffold]
+/// widget. The [NavigationBar] has three [NavigationDestination] widgets
+/// and the [selectedIndex] is set to index 0. The `onDestinationSelected` callback
+/// changes the selected item's index and displays a corresponding widget in the body of the [Scaffold].
+///
+/// ** See code in examples/api/lib/material/navigation_bar.dart **
```
Comment (werainkhatri) [inline packages/flutter/lib/src/material/navigation_bar.dart:42]:
```suggestion
/// ** See code in examples/api/lib/material/navigation_bar/navigation_bar.dart **
```

**Review by werainkhatri — DISMISSED** [review]

File: packages/flutter/lib/src/material/navigation_bar.dart (hunk 2 above)
Comment (maheshmnj) [reply packages/flutter/lib/src/material/navigation_bar.dart]: Ah thanks, that was a silly mistake.

File: examples/api/lib/material/navigation_bar/navigation_bar.dart [hunk 3]
```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for NavigationBar
+
+import 'package:flutter/material.dart';
```
Comment (HansMuller) [inline examples/api/lib/material/navigation_bar/navigation_bar.dart:7]: There are some trivial formatting issues here and the code can be simplified a little 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({ Key? key }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: NavigationExample());
  }
}

class NavigationExample extends StatefulWidget {
  const NavigationExample({Key? key}) : super(key: key);

  @override
  State<NavigationExample> createState() => _NavigationExampleState();
}

class _NavigationExampleState extends State<NavigationExample> {
  int currentPageIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      bottomNavigationBar: NavigationBar(
        onDestinationSelected: (int index) {
          setState(() {
            currentPageIndex = index;
          });
        },
        selectedIndex: currentPageIndex,
        destinations: const <Widget>[
          NavigationDestination(
            icon: Icon(Icons.explore),
            label: 'Explore',
          ),
          NavigationDestination(
            icon: Icon(Icons.commute),
            label: 'Commute',
          ),
          NavigationDestination(
            selectedIcon: Icon(Icons.bookmark),
            icon: Icon(Icons.bookmark_border),
            label: 'Saved',
          ),
        ],
      ),
      body: <Widget>[
        Container(
          color: Colors.red,
          alignment: Alignment.center,
          child: const Text('Page 1'),
        ),
        Container(
          color: Colors.green,
          alignment: Alignment.center,
          child: const Text('Page 2'),
        ),
        Container(
          color: Colors.blue,
          alignment: Alignment.center,
          child: const Text('Page 3'),
        ),
      ][currentPageIndex],
    );
  }
}
```

**Review by HansMuller — COMMENTED** [review]
The example will need a test, see examples/api/test

**Comment by maheshmnj** [comment]
> The example will need a test, see examples/api/test

Thanks for reviewing, I have added basic tests. let me know if you want me to add more scenarios.

File: examples/api/lib/material/navigation_bar/navigation_bar.dart (hunk 3 above)
Comment (maheshmnj) [reply examples/api/lib/material/navigation_bar/navigation_bar.dart:7]: Please check I have made some changes to the sample code.

Comment (HansMuller) [reply examples/api/lib/material/navigation_bar/navigation_bar.dart:7]: I had reformatted your example per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo. That was the comment (above) was.  Please try and align your updated example with what I'd written.

**Review by HansMuller — CHANGES_REQUESTED** [review]

File: examples/api/lib/material/navigation_bar/navigation_bar.dart (hunk 3 above)
Comment (maheshmnj) [reply examples/api/lib/material/navigation_bar/navigation_bar.dart:7]: Thanks, I have formatted the sample code as per above

**Comment by HansMuller** [comment]
@maheshmnj - the test failures aren't related to this PR. Please sync with master and push an empty commit to get the tests to run again.

**Comment by maheshmnj** [comment]
Hi, just checking if there is anything on my side to fix 1 failing checkmark.

**Comment by HansMuller** [comment]
Per @dnfield - the source of the problem is likely the logic used to pick a commit for customer testing. A merge will not fix the issue, this PR needs to be rebased.

**Comment by maheshmnj** [comment]
Oops! it's still failing here's what I did
```
git checkout master 
git pull remote master
git checkout navigation-bar-docs
git rebase master
/// resolved conflicts
git add .
git rebase --continue
git push  /// failed
git push -f
```

Let me know if I missed anything

Thanks

**Comment by dnfield** [comment]
@maheshmnj - I think something like this should work:

```
git fetch remote
git rebase remote/master
# resolve conflicts
git push -f
```

Alternatively, just start a fresh PR with these changes that's based on the most recent commit from master.

The only other time I've seen these two failures (on flutter_svg and rfw) it was related to this issue. The branch was forked from a commit behind where the customer testing was updated to account for golden file changes.

**Comment by dnfield** [comment]
This PR is still getting a relatively old commit of flutter/tests - https://github.com/flutter/tests/commit/1943e74d1f88083797a9b5275d1f825e0e1d47cf - a properly done rebase should resolve that, but sometimes git is finnicky about that. If another rebase attempt does not fix this I would just close the PR and open a new one with these changes applied on a fresh branch.

**Comment by HansMuller** [comment]
My git skills are limited and so I have to admit that I'd take Dan's advice about just creating a fresh PR - https://github.com/flutter/flutter/pull/97046#issuecomment-1054480584. Sometimes that's the easiest way to deal with this kind of thing.

**Comment by maheshmnj** [comment]
Thanks for the suggestion @dnfield  luckily this worked out.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**